pkg/jobs/jobs.go

```diff
@@ -32,6 +33,33 @@ import (
 	"github.com/cockroachdb/redact"
 )
 
+// jobDumpTraceMode is the type that represents the mode in which a traceable
+// job will dump a trace zip.
+type jobDumpTraceMode int64
+
+const (
+	// A Traceable job will not dump a trace zip.
+	noDump jobDumpTraceMode = iota
+	// A Traceable job will dump a trace zip on failure.
+	dumpOnFail
+	// A Traceable job will dump a trace zip in any of paused, canceled, failed,
+	// succeeded states.
+	dumpOnStatusChange
+)
+
+var traceableJobDumpTraceMode = settings.RegisterEnumSetting(
+	"jobs.trace.force_dump_mode",
+	"determines the state in which all traceable jobs will dump their cluster wide, inflight, "+
+		"trace recordings. Traces may be dumped never, on fail, "+
+		"or on any status change i.e paused, canceled, failed, succeeded.",
+	"never",
+	map[int64]string{
+		int64(noDump):             "never",
+		int64(dumpOnFail):         "onFail",
+		int64(dumpOnStatusChange): "onStatusChange",
```

Comment:
Sorry for all the bikeshedding, but I think that I prefer `onStop` based on our offline discussion.